### PR TITLE
Emc/fix/1987-missing-projects-in-v2-0-2-api

### LINKED ISSUE #1987: [api] Missing Projects from staging & production
Over 150 projects are missing from the search results after moving from v2.0.0 to v2.0.2 of the API

https://v2.0.0.api.carbonmark.com/projects
https://v2.0.2.api.carbonmark.com/projects

e.g VCS-1100 & VCS-1162

### PR DESCRIPTION
**Pull request by 0xemc**
## Description

There were two issues introduced to the API since 2.0.0

1. the structure of `carbonProjects` is slightly different in that there used to be a seperate entry each vintage. Now there is a single entry with "carbonCredits" as objects within each entry that represent each vintage.

We were only grabbing the first in the array.

2. The hard coded values for the pool addresses contained uppercase characters and somewhere along the line the `isValidProject`  utility was refactored and no longer converted  to lowercase before comparison.

## Related Ticket

Closes #1987

## Notes For QA

Please check that the number of projects is 354 and that VCS-1100 and VCS-1162 are returned when searched for

Specific pages, components or journeys that might be affected:
- 

Relevant preview URLs:
- 

Other notes:
-

### CODE REVIEW

File: carbonmark-api/src/routes/projects/get.ts [hunk 1]
```diff
@@ -74,27 +74,27 @@ const handler = (fastify: FastifyInstance) =>
...
-        standard,
-        registryProjectId,
-        vintage: project.carbonCredits[0].vintage.toString(),
+      project.carbonCredits.forEach((credit) => {
+        if (!isValidPoolProject(project)) {
+          console.debug(
+            `Project with id ${project.projectID} is considered invalid due to a balance of zero across all tokens and has been filtered`
+          );
+          return;
+        }
```
Comment (biwano) [inline carbonmark-api/src/routes/projects/get.ts:83]: This does not depend on credit. You could put it in the project loop.

File: carbonmark-api/src/routes/projects/get.ts [hunk 2]
```diff
@@ -74,27 +74,27 @@ const handler = (fastify: FastifyInstance) =>
...
+        if (!CreditId.isValidProjectId(project.projectID)) {
+          console.debug(
+            `Project with id ${project.projectID} is considered to have an invalid id and has been filtered`
+          );
+          return;
+        }
+        const [standard, registryProjectId] = project.projectID.split("-");
```
Comment (biwano) [inline carbonmark-api/src/routes/projects/get.ts:90]: Same

**Review by biwano — APPROVED** [review]
Just a little optimization request if you have time.

**Review by jabby09 — COMMENTED** [review]
@0xemc 

Number of projects is 343, closer to 354 but not quite there.

VCS-1100 shows up 3 times
Screenshot 2023-12-06 at 10 38 53 AM

VCS-1162 shows up once.

**Pull request merged**